### ISSUE TITLE
Turn gradient_accumulate into a separate aten op

### ISSUE THREAD

**Issue by Chillee** [pytorch/pytorch#154767]
### 🚀 The feature, motivation and pitch

There's a variety of cases where you might want to have special semantic information about the gradient accumulation operation. One example is https://aclanthology.org/2023.acl-long.669.pdf, where they perform autodiff over the (max, x) field as opposed to the standard (+, x) field. I believe this could be implemented pretty straightforwardly in PyTorch if the gradientAccumulate dispatched to a special op (say `aten::grad_add`) as opposed to `aten::add`.

It could also be useful for other purposes, such as tracking gradient accumulation for tensor subclasses.

**Comment by ngimel** [comment]
https://dev-discuss.pytorch.org/t/implementing-generalized-backpropagation/1875/5

**Comment by Chillee** [comment]
In my case, the main motivation prompting this issue is moreso being able to specially handle grad accumulate for other things like rounding or tensor subclasses.

**Comment by zou3519** [comment]
Btw the operator likely needs to be CompositeImplicitAutograd, otherwise there may be BC implications for Tensor subclasses

**Comment by zou3519** [comment]
@albanD says:

Do you want to change accumulate_grad, or do you also want to change what happens when someone uses a Tensor multiple times in the forward (this turns into multiple adds in the backward, but not necessarily an accumulate_grad op, if it's an intermediate)
